### PR TITLE
fix(rp2040): Moved subs and adds code to a dedicated function

### PR DESCRIPTION
**Pull request by Turro75**
ADCS, ADDS T1/T2, ADDSreg, CMN all use additionUpdateFlags

CMP, CMPreg T1/T2, NEGS/RSBS, SBCS, SUBS T1/T2, SUBSreg all use substractUpdateFlags

### CODE REVIEW

File: src/rp2040.ts [hunk 1]
```diff
@@ -880,6 +880,28 @@ export class RP2040 {
     }
   }
 
+  substractUpdateFlags(minuend: number, subtrahend: number) {
```
Comment (urish) [inline src/rp2040.ts:883]: Funny, I didn't know minuend/subtrahend was even a word! I learned something new :)

Can we make this method private?

File: src/rp2040.ts [hunk 2]
```diff
@@ -880,6 +880,28 @@ export class RP2040 {
...
+    this.N = !!(result & 0x80000000);
+    this.Z = (result & 0xffffffff) === 0;
+    this.C = minuend >= subtrahend;
+    this.V =
+      (!!(result & 0x80000000) && !(minuend & 0x80000000) && !!(subtrahend & 0x80000000)) ||
+      (!(result & 0x80000000) && !!(minuend & 0x80000000) && !(subtrahend & 0x80000000));
+    return result;
+  }
+
+  additionUpdateFlags(addend1: number, addend2: number) {
```
Comment (urish) [inline src/rp2040.ts:894]: Let's call it `addUpdateFlags()` (because the other one is subtract)

**Review by urish — CHANGES_REQUESTED** [review]
I love how it makes the code shorter! Let's just make the new methods private? (+ rename additionUpdateFlags to addUpdateFlags)

File: src/rp2040.ts (hunk 1 above)
Comment (Turro75) [reply src/rp2040.ts]: > I didn't know minuend/subtrahend was even a word!
the google's power :-)

**Comment by urish** [comment]
Great, thanks a lot valerio! 

Just published it as  rp2040js 0.6.3

**Pull request merged**